python/lsst/cp/pipe/ptc.py

```diff
@@ -1103,39 +1107,56 @@ def _plotPtc(self, dataset, ptcFitType, pdfPages):
+                minMeanVecOriginal = np.min(meanVecOriginal)
+                maxMeanVecOriginal = np.max(meanVecOriginal)
+                deltaXlim = maxMeanVecOriginal - minMeanVecOriginal
+
+                a.plot(meanVecFit, ptcFunc(pars, meanVecFit), color='red')
+                a.plot(meanVecFinal, pars[0] + pars[1]*meanVecFinal, color='green', linestyle='--')
+                a.scatter(meanVecFinal, varVecFinal, c='blue', marker='o', s=markerSize)
+                a.scatter(meanVecOutliers, varVecOutliers, c='magenta', marker='s', s=markerSize)
+                a.set_xlabel(r'Mean signal ($\mu$, DN)', fontsize=labelFontSize)
+                a.set_xticks(meanVecOriginal)
+                a.set_ylabel(r'Variance (DN$^2$)', fontsize=labelFontSize)
+                a.tick_params(labelsize=11)
+                a.text(0.03, 0.8, stringLegend, transform=a.transAxes, fontsize=legendFontSize)
+                a.set_xscale('linear', fontsize=labelFontSize)
+                a.set_yscale('linear', fontsize=labelFontSize)
+                a.set_title(amp, fontsize=titleFontSize)
+                a.set_xlim([minMeanVecOriginal - 0.2*deltaXlim, maxMeanVecOriginal + 0.2*deltaXlim])
+
+                # Same, but in log-scale
+                a2.plot(meanVecFit, ptcFunc(pars, meanVecFit), color='red')
+                a2.scatter(meanVecFinal, varVecFinal, c='blue', marker='o', s=markerSize)
+                a2.scatter(meanVecOutliers, varVecOutliers, c='magenta', marker='s', s=markerSize)
+                a2.set_xlabel(r'Mean Signal ($\mu$, DN)', fontsize=labelFontSize)
+                a2.set_ylabel(r'Variance (DN$^2$)', fontsize=labelFontSize)
+                a2.tick_params(labelsize=11)
+                a2.text(0.03, 0.8, stringLegend, transform=a2.transAxes, fontsize=legendFontSize)
+                a2.set_xscale('log')
+                a2.set_yscale('log')
+                a2.set_title(amp, fontsize=titleFontSize)
+                a2.set_xlim([minMeanVecOriginal, maxMeanVecOriginal])
+            else:
+                a.set_xlabel(r'Mean signal ($\mu$, DN)', fontsize=labelFontSize)
+                a.set_ylabel(r'Variance (DN$^2$)', fontsize=labelFontSize)
+                a.tick_params(labelsize=11)
+                a.text(0.03, 0.8, stringLegend, transform=a.transAxes, fontsize=legendFontSize)
+                a.set_xscale('linear', fontsize=labelFontSize)
+                a.set_yscale('linear', fontsize=labelFontSize)
+                a.set_title(f"{amp} (BAD)", fontsize=titleFontSize)
+
+                a2.set_xlabel(r'Mean Signal ($\mu$, DN)', fontsize=labelFontSize)
+                a2.set_ylabel(r'Variance (DN$^2$)', fontsize=labelFontSize)
+                a2.tick_params(labelsize=11)
+                a2.text(0.03, 0.8, stringLegend, transform=a2.transAxes, fontsize=legendFontSize)
+                a2.set_xscale('log')
+                a2.set_yscale('log')
+                a2.set_title(f"{amp} (BAD)", fontsize=titleFontSize)
```

Comment:
I think pulling out the elements in common and calling those together outside of the if/else would help keep things together and easier to understand & maintain.